docs/guides/js-waku/debug-waku-dapp.md

```diff
@@ -0,0 +1,99 @@
+---
+title: How to Debug Your Waku DApp and WebSocket
+---
+
+This guide provides detailed steps to enable and use debug logs to troubleshoot your Waku DApp, whether in a NodeJS or browser environment and check your WebSocket connections in [nwaku](/guides/run-nwaku-node).
+
+## Enabling Debug Logs
+
+When resolving issues in your Waku DApp, debug logs can be helpful. The `@waku/sdk` and `libp2p` packages use the debug tool to handle and show logs that help you debug effectively.
+
+### NodeJS Environments
+
+To enable debug logs for `@waku/sdk` on NodeJS, you must set the `DEBUG` environment variable. To only enable debug logs for `@waku/sdk`:
+
+```shell
+export DEBUG=waku*
+```
+
+To enable debug logs for both `@waku/sdk` and `libp2p`:
+
+```shell
+export DEBUG=waku*,libp2p*
+```
+
+To enable debug logs for all components:
+
+```shell
+export DEBUG=*
+```
+
+### Browser Environments
+
+To view debug logs in your browser's console, modify the local storage and add the `debug` key. Here are guides for various modern browsers:
+
+- [Firefox](https://firefox-source-docs.mozilla.org/devtools-user/storage_inspector/local_storage_session_storage/index.html)
+- [Google Chrome](https://developer.chrome.com/docs/devtools/storage/localstorage/)
+
+| KEY | VALUE | DESCRIPTION |
+| - | - | - |
+| `debug` | `waku*` | Enables `@waku/sdk` debug logs |
+| `debug` | `waku*,libp2p*` | Enables `@waku/sdk` and `libp2p` debug logs |
+| `debug` | `*` | Enables all debug logs |
+
+## Checking WebSocket Setup
+
+[Nwaku](/guides/run-nwaku-node) provides native support for WebSocket (`ws`) and WebSocket Secure (`wss`) protocols. These are the only [transports](/overview/concepts/transports) supported for connecting to the Waku Network via browsers.
+
+It's important to note that browsers impose certain limitations on WebSocket usage:
+
+- **Secure Context Requirement**: Insecure subroutines are prohibited in secure contexts. This means that on an `https://` webpage, only `wss` connections are permitted, while `ws` connections are not allowed.
+- **Certificate Validation**: Certificate validation rules are consistent for `https` and `wss` connections. Certificates must not be expired, issued by a recognized Certificate Authority (CA), and match the domain name, among other criteria.
+- **User Feedback on Errors**: Web browsers do not display errors related to subroutines to the user. If a WebSocket connection encounters an issue, users won't be alerted directly; you'll need to check the browser's console for error details.
+
+:::info
+The mentioned restrictions are not applicable if the webpage is served locally, like on `localhost` or `127.0.0.1`.
```

Comment:
It's only the _secure context requirement_ one that does not apply when served locally.